.github/scripts/validate_new_chain_request.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+
+"""
+This script is used to validate a GitHub issue for a new chain request.
+What it does:
+1. Extracts the RPC URL from the issue body
+2. Extracts the chain ID from the RPC URL
+3. Checks if the chain ID is in chainlist
+4. Checks if the factory is already deployed
+5. Checks if the deployer address is pre-funded
+
+If any of the above checks fails, the script will exit with an error code and set an environment variable
+`COMMENT_OUTPUT` with an error message. The GitHub action will then use this variable to post a comment.
+For the above reason, the script needs to be executed in the GitHub's action shell, for example:
+```
+. ./.github/scripts/validate_new_chain_request.sh
+```
+
+The script expects the following environment variables to be set:
+- `ISSUE_BODY` - the body of the issue
+- `FACTORY_ADDRESS` - the address of the factory contract
+- `FACTORY_DEPLOYER_ADDRESS` - the address of the factory deployer
+- `FACTORY_BYTECODE` - the bytecode of the factory contract
+
+"""
+
+ERROR_MSG_RPC_FAILURE="""
+**⛔️ Error:**<br>
+Could not fetch the chain ID from the RPC URL.<br>
+Either the RPC URL is missing, or the RPC is not responding.<br>
+**Make sure your issue follows the issue template and includes the RPC URL.**<br>
+:sparkles: You can edit the issue to trigger the check again. :sparkles:"""
+ERROR_MSG_CHAINLIST_FAILURE="""
+**⛔️ Error:**<br>
+Couldn't find the chain in the chainlist. Please make sure it's added to the chainlist.<br>
+For more information on how to add a chain, please refer to the [chainlist repository](https://github.com/ethereum-lists/chains).<br>
+:sparkles: You can edit the issue to trigger the check again. :sparkles:"""
+ERROR_MSG_PREFUND_CHECK="""
+**⛔️ Error:**<br>
+There was an error while estimating the deployment transaction and checking the pre-fund. Please make sure that the RPC URL is valid and reachable.<br>
+:sparkles: You can edit the issue to trigger the check again. :sparkles:"""
+SUCCESS_MSG="""
+**✅ Success:**<br>
+The issue description is valid:<br>
+- The RPC URL is valid<br>
+- The chain is in the chainlist<br>
+- The deployer address is pre-funded<br>
+:sparkles: The team will be in touch with you soon :sparkles:"""
+
+
+trim() {
+    # 3.5.3 Shell Parameter Expansion at https://www.gnu.org/software/bash/manual/bash.html
+    local var="$*"
+    # remove leading whitespace characters
+    var="${var#"${var%%[![:space:]]*}"}"
+    # remove trailing whitespace characters
+    var="${var%"${var##*[![:space:]]}"}"
+    echo "$var"
+}
+
+rpc_url=$(echo $ISSUE_BODY | egrep -o 'https?://[^ ]+' -m 1 | head -1)
```

Comment:
```suggestion
rpc_url="$(echo $ISSUE_BODY | egrep -o 'https?://[^ ]+' -m 1)"
```

I _think_ `head` is superfluous here since `-m 1` already stops after the first match. Local testing seems to confirm this (taken from an issue):

```sh
$ egrep -o 'https?://[^ ]+' -m 1 <<TEXT
## Network information

- Name: Rootstock Testnet
- ChainId: 31
- RPC http/s address: https://public-node.testnet.rsk.co
- Link to https://chainlist.org/chain/31
- (Optional) Blockexplorer: https://explorer.testnet.rsk.co
- [x] I have run RPC='https://public-node.testnet.rsk.co' yarn estimate
{ chainId: 31 }
{ estimate: '72291' }
{ gasPriceGwei: '0.065164', gasPrice: '65164000' }
{ requiredFunds: '0.000004710770724' }
- [x] I have [sent](https://explorer.testnet.rsk.co/tx/0x712d7ecbcb3a0f67f26194340db636e3805cf3e177c0f3ee1ffbac5038cb3810) requiredFunds to [0xE1CB04A0fA36DdD16a06ea828007E35e1a3cBC37](https://explorer.testnet.rsk.co/address/0xe1cb04a0fa36ddd16a06ea828007e35e1a3cbc37)
TEXT

https://public-node.testnet.rsk.co
```

Also, I would quote it here.

Reply:
I don't remember the exact reason why I added it, but I remember landing on this StackOverflow question: https://stackoverflow.com/a/69833537/7820085

Reply:
TIL! It's an edge case if there is more than one match on a single line. I don't think its possible with the current pattern, but its better to leave it in and be safe.

Reply:
It seems to work. I removed the head

Reply:
Ah, somehow, when I answered you, the message with "leave it and be safe" wasn't loaded. Will bring it back